src/assets/wise5/classroomMonitor/classroomMonitorComponents/peer-group/peer-group-dialog/peer-group-dialog.component.ts

```diff
@@ -0,0 +1,209 @@
+  componentLabel: string;
+  componentTypeToLabel: any = {
+    PeerChat: 'Peer Chat'
+  };
+  currentPeriodChangedSubscription: Subscription;
+  groupings: any[] = [];
+  nextAvailableGroupId: number = 1;
+  stepTitle: string;
+  unassignedWorkgroups: any[] = [];
+  workgroups: any[] = [];
+
+  constructor(
+    private ConfigService: ConfigService,
+    @Inject(MAT_DIALOG_DATA) public data: any,
+    private PeerGroupService: PeerGroupService,
+    private ProjectService: ProjectService,
+    private TeacherDataService: TeacherDataService
+  ) {
+    this.componentId = data.componentId;
+    this.nodeId = data.nodeId;
+    this.periodId = data.periodId;
+  }
+
+  ngOnInit() {
+    this.subscribeToPeriodChanged();
+    this.componentLabel = this.componentTypeToLabel[
+      this.ProjectService.getComponentType(this.nodeId, this.componentId)
+    ];
+    this.stepTitle = this.ProjectService.getNodePositionAndTitleByNodeId(this.nodeId);
+    this.PeerGroupService.retrieveGroupings(this.nodeId, this.componentId).subscribe(
+      (groupings: any) => {
+        // TODO
+      },
+      () => {
+        this.workgroups = this.getWorkgroupsInPeriod();
+        this.setDummyGroupingsAndUnassignedWorkgroups();
+      }
+    );
+  }
+
+  subscribeToPeriodChanged(): void {
+    this.currentPeriodChangedSubscription = this.TeacherDataService.currentPeriodChanged$.subscribe(
+      ({ currentPeriod }) => {
+        this.periodId = currentPeriod.periodId;
+        this.workgroups = this.getWorkgroupsInPeriod();
+        this.setDummyGroupingsAndUnassignedWorkgroups();
+      }
+    );
+  }
+
+  getWorkgroupsInPeriod(): any[] {
+    return this.ConfigService.getWorkgroupsByPeriod(this.periodId).filter(
+      (workgroup) => workgroup.workgroupId != null
+    );
+  }
+
+  ngOnDestroy() {
+    this.currentPeriodChangedSubscription.unsubscribe();
+  }
+
+  setDummyGroupingsAndUnassignedWorkgroups(): void {
+    this.groupings = [];
+    this.unassignedWorkgroups = [];
+    const numUnassignedWorkgroups = this.getDummyNumUnassignedWorkgroups(this.workgroups);
+    let tempWorkgroups = [];
+    for (let w = 0; w < this.workgroups.length - numUnassignedWorkgroups; w++) {
+      tempWorkgroups.push(this.workgroups[w]);
+      if (tempWorkgroups.length >= 2) {
+        this.addGrouping(this.createGrouping(this.nextAvailableGroupId++, tempWorkgroups));
+        tempWorkgroups = [];
+      }
+    }
+    for (
+      let w = this.workgroups.length - numUnassignedWorkgroups;
+      w < this.workgroups.length;
+      w++
+    ) {
+      this.unassignedWorkgroups.push(this.workgroups[w]);
+    }
+  }
+
+  getDummyNumUnassignedWorkgroups(workgroups: any[]): number {
+    if (workgroups.length >= 4) {
+      return 2;
+    } else if (workgroups.length === 3) {
+      return 1;
+    } else {
+      return 0;
+    }
+  }
+
+  createGrouping(id: number = this.nextAvailableGroupId, workgroups: any[]): any {
+    return {
+      id: id,
+      workgroups: workgroups
+    };
+  }
```

Comment:
Instead of returning any, should we define a Grouping class and return that?

Reply:
This is code used for mocking and will be deleted later. We should create a PeerGroup class when we integrate with the real backend calls.